cmdx/docs.go

```diff
@@ -0,0 +1,39 @@
+package cmdx
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+)
+
+// GenerateMarkdownTree generate cobra cmd commands tree as markdown file
+// rootOutputPath determines the folder where the markdown files are written
+func GenerateMarkdownTree(rootOutputPath string, cmd *cobra.Command) error {
```

Comment:
@mabdh do we want to generate a single markdown or multiple?

Reply:
I think multiple markdowns would be easier to navigate via sidebar, wdyt @ravisuhag ?

Reply:
<img width="1744" alt="Screen Shot 2022-10-19 at 13 24 04" src="https://user-images.githubusercontent.com/4079709/196612801-d0b895e2-0ade-48f8-979b-8ae7982ea8d0.png">

this is how the markdown shown in Siren

Reply:
If we are showing it reference i think it better to show it single markdown. How about we make this as an option? Support both in salt.

Reply:
Yes fair enough, we could also do that.